### ISSUE TITLE
Checks Summary Tab not refreshing

### ISSUE THREAD

**Issue by SQL-Snowball99** [trimble-oss/dba-dash#436]
Hi David,

After upgrading to 2.25.0, the Summary Tab  for Checks for all instances is not refreshing. You can only see the summary on the per instance level. I restarted the service and still having the issue . 

Is it possible to also auto-refresh the page? e.g It can be every 5 minutes. 
![image](https://user-images.githubusercontent.com/104442916/201049505-94be82a7-acb8-4169-ae1d-5f013e3c8e40.png)

Best Regards

**Comment by DavidWiseman** [comment]
That's odd.  Maybe there is a performance issue running the query to get the summary data.  What happens if you run this against the repository DB (similar to what the app will run):

```sql
DECLARE @InstanceIDs VARCHAR(MAX)
SELECT @InstanceIDs=STUFF((SELECT CONCAT(',', InstanceID)
					FROM dbo.Instances 
					WHERE IsActive=1
					FOR XML PATH(''),TYPE).value('.','VARCHAR(MAX)'),1,1,'')

exec dbo.Summary_Get @InstanceIDs=@InstanceIDs,@IncludeHidden=0
```

If you wanted to create an extended events session to capture the queries run from the GUI you can use:

```sql
CREATE EVENT SESSION [DBADashGUI] ON SERVER 
ADD EVENT sqlserver.rpc_completed(
    WHERE ([sqlserver].[client_app_name]=N'DBADashGUI')),
ADD EVENT sqlserver.sql_batch_completed(
    ACTION(sqlserver.client_app_name)
    WHERE ([sqlserver].[client_app_name]=N'DBADashGUI'))
WITH (MAX_MEMORY=4096 KB,EVENT_RETENTION_MODE=ALLOW_SINGLE_EVENT_LOSS,MAX_DISPATCH_LATENCY=30 SECONDS,MAX_EVENT_SIZE=0 KB,MEMORY_PARTITION_MODE=NONE,TRACK_CAUSALITY=OFF,STARTUP_STATE=OFF)
GO
```

The GUI doesn't interact with the service in any way so a restart of the service won't help.  The GUI only queries the repository DB.  Restarting the service is a way to trigger the collections to run though - instead of waiting for then next scheduled run time.

Let me know if you find anything.  The Summary page is working for me on 2.25.0.

**Comment by SQL-Snowball99** [comment]
Hi, 
I get this error when executing the code:
![image](https://user-images.githubusercontent.com/104442916/201092924-07bdb995-ec19-4285-81ad-a378cec83024.png)

**Comment by SQL-Snowball99** [comment]
I think the issue is with the user tags (InstanceRole in my case) created whereby I did not assign some instances to these tags. I will assign the tag to all the instances and advise if it resolves my issue.

**Comment by SQL-Snowball99** [comment]
This is the query throwing the error:
![image](https://user-images.githubusercontent.com/104442916/201178931-36fe3fb0-ed1d-46bb-b846-a90fff64875f.png)

Thanks.

**Comment by DavidWiseman** [comment]
Thanks.  That was one of the possibilities I found.  It looks like you have a drive reporting capacity/size of 0 which is a bit odd.  If you want to apply a hotfix, you can run the ALTER VIEW statement below, which should fix your issue.  I'll get this included in the next release.

```sql
ALTER VIEW dbo.DriveStatus
AS
SELECT I.InstanceID,
	D.DriveID,
	I.Instance,
	I.InstanceDisplayName,
	I.ConnectionID,
	D.Name,
	D.Label,
	D.Capacity/POWER(1024.0,3) TotalGB,
	D.UsedSpace/POWER(1024.0,3) UsedGB,
	D.FreeSpace/POWER(1024.0,3) FreeGB,
	D.FreeSpace/CAST(NULLIF(D.Capacity,0) AS DECIMAL) AS PctFreeSpace,
	DATEDIFF(mi,SSD.SnapshotDate,GETUTCDATE()) AS SnapshotAgeMins,
	SSD.SnapshotDate,
	CASE WHEN cdt.WarningThreshold IS NULL AND cdt.CriticalThreshold IS NULL THEN 3 WHEN DATEDIFF(mi,SSD.SnapshotDate,GETUTCDATE()) > cdt.CriticalThreshold THEN 1 WHEN DATEDIFF(mi,SSD.SnapshotDate,GETUTCDATE()) > cdt.WarningThreshold THEN 2 ELSE 4 END AS SnapshotStatus,
	cfg.DriveWarningThreshold,
	cfg.DriveCriticalThreshold,
	cfg.DriveCheckType,
	chk.Status,
	CASE chk.Status WHEN 1 THEN 'Critical' WHEN 2 THEN 'Warning' WHEN 3 THEN 'N/A' WHEN 4 THEN 'OK' END AS StatusDescription,
	CASE WHEN cfg.InstanceID = -1 AND cfg.DriveID = -1 THEN 'Root'
	WHEN cfg.DriveID =-1 THEN 'Instance'
	WHEN cfg.InstanceID IS NULL THEN 'None'
	ELSE 'Drive' END DriveCheckConfiguredLevel,
	CASE WHEN cfg.DriveID=-1 OR cfg.InstanceID=-1 THEN CAST(1 AS BIT) ELSE CAST(0 AS BIT) END AS IsInheritedThreshold
FROM dbo.Drives D
INNER JOIN dbo.Instances I ON D.InstanceID = I.InstanceID
INNER JOIN dbo.CollectionDates SSD ON SSD.InstanceID = I.InstanceID AND SSD.Reference='Drives'
OUTER APPLY(SELECT TOP(1) T.* 
			FROM dbo.DriveThresholds T 
			WHERE (D.InstanceID = T.InstanceID OR T.InstanceID = -1)
			AND (D.DriveID = T.DriveID  OR T.DriveID = -1)
			ORDER BY InstanceID DESC,DriveID DESC
			) cfg
OUTER APPLY(SELECT TOP(1) T.WarningThreshold, T.CriticalThreshold
			FROM dbo.CollectionDatesThresholds T
			WHERE T.InstanceID = D.InstanceID OR T.InstanceID=-1
			AND T.Reference = 'Drives'
			ORDER BY T.InstanceID DESC) cdt
OUTER APPLY(SELECT CASE WHEN cfg.DriveCheckType = '%' 
		AND FreeSpace/CAST(NULLIF(Capacity,0) AS DECIMAL) < cfg.DriveCriticalThreshold THEN 1
		WHEN cfg.DriveCheckType = '%' 
		AND FreeSpace/CAST(NULLIF(Capacity,0) AS DECIMAL) < cfg.DriveWarningThreshold THEN 2
		WHEN cfg.DriveCheckType = 'G' 
		AND FreeSpace/POWER(1024.0,3)< cfg.DriveCriticalThreshold THEN 1
		WHEN cfg.DriveCheckType = 'G' 
		AND FreeSpace/POWER(1024.0,3) < cfg.DriveWarningThreshold THEN 2
		WHEN cfg.DriveWarningThreshold IS NULL AND cfg.DriveCriticalThreshold IS NULL THEN 3
		ELSE 4 END AS Status) chk
WHERE D.IsActive=1
AND I.IsActive=1
```

**Comment by DavidWiseman** [comment]
PS Next version will display an error message in case of an error. Hopefully, you won't see this screen very often:
![image](https://user-images.githubusercontent.com/33157668/201201230-743e6576-65f0-427d-91ab-df04a50d0512.png)

**Comment by SQL-Snowball99** [comment]
Hi David, 
The hotfix did the trick and the issue is now resolved.

Thank you.

**Comment by DavidWiseman** [comment]
That's great - thanks for letting me know.  The fix has already been done and will be included in the next release.

**Comment by DavidWiseman** [comment]
[Fixed in 2.26.0](https://dbadash.com/blog/whats-new-in-2.26.0/) 🚀
